src/Draco.Compiler/Internal/Syntax/Lexer.cs

```diff
@@ -496,6 +496,31 @@ private Unit LexString()
         // Check for escape sequence
         if (ch == '\\')
         {
+            // End of file after the escape
+            // Pop the string mode so the next token is End of input
+            if (this.Peek(offset + mode.ExtendedDelims + 1) == '\0')
+            {
+                this.PopMode();
+
+                var untilEndOffset = 0;
+                while (this.Peek(offset + untilEndOffset) != '\0')
```

Comment:
This condition together with the condition of the if-statement is a bit fuzzy to me, this while loop might be wrong or the fix itself should not happen here directly. Maybe, when we are in an interpolation mode and find the end of input, we should just pop off everything and return end of input there?